test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -88,4 +99,95 @@ static public void testPassingNullToIsFlattenedThrowsException() throws Throwabl
 			myUnsafe.isFlattened(null);
 		});
 	}
+
+	@Test
+	static public void testValueHeaderSizeOfValueType() throws Throwable {
+		long headerSize = myUnsafe.valueHeaderSize(ValueTypePoint2D.class);
+		if (isCompressedRefsEnabled) {
+			assertEquals(headerSize, 4);
+		} else {
+			assertEquals(headerSize, 8);
+		}
+	}
+
+	@Test
+	static public void testValueHeaderSizeOfClassWithLongField() throws Throwable {
+		long headerSize = myUnsafe.valueHeaderSize(ValueTypeWithLongField.class);
+		if (isCompressedRefsEnabled && !isFlatteningEnabled) {
+			assertEquals(headerSize, 4);
+		} else {
+			// when the class has an 8 byte field, 4 bytes of padding are added to the class. These padding bytes are included in the headerSize
```

Comment:
This comment is still not correct. It should be:

"In compressed refs mode, when the class has 8 byte field(s) only, 4 bytes of padding are added to the class. These padding bytes are included in the headerSize."

Reply:
ah, that makes more sense. Before when you said "only" I thought you meant that compressed refs weren't required. It's been updated in 71981a8